Add ESA CCI LST cmorizer and diagnostic

morobking opened the pull request:
<!---
Please do not delete this text completely, but read the text below and keep items that seem relevant.
If in doubt, just keep everything and add your own text at the top.
--->

Before you start, please read our [contribution guidelines](https://docs.esmvaltool.org/en/latest/community/introduction.html).

Please discuss your idea with the development team before getting started, to avoid disappointment later. The way to do this is to open a new issue on GitHub. If you are planning to modify an existing functionality, please discuss it with the original author(s) by tagging them in the issue.

* * *
This pull request has been requested by @valeriupredoi to look at the CMORizing of the ESA CCI LST data.
Issue #1896 is being used to document this work and the recipe and diagnostic it will also produce.

**Tasks**

-   [ ] [Create an issue](https://github.com/ESMValGroup/ESMValTool/issues) to discuss what you are going to do, if you haven't done so already (and add the link at the bottom)
-   [x] Give this pull request a descriptive title that can be used as a one line summary in a changelog
-   [ ] Make sure your code is composed of functions of no more than 50 lines and uses meaningful names for variables
-   [x] Circle/CI tests pass. Status can be seen below your pull request. If the tests are failing, click the link to find out why.
-   [ ] Preferably Codacy code quality checks pass, however a few remaining hard to solve Codacy issues are still acceptable. Status can be seen below your pull request. If there is an error, click the link to find out why. If you suspect Codacy may be wrong, please ask by commenting.
-   [x] Please use `yamllint` to check that your YAML files do not contain mistakes
-   [ ] (Only if really necessary) Add any additional dependencies needed for the diagnostic script to setup.py, esmvaltool/install/R/r_requirements.txt or esmvaltool/install/Julia/Project.toml (depending on the language of your script) and also to package/meta.yaml for conda dependencies (includes Python and others, but not R/Julia)
-   [ ] If new dependencies are introduced, check that the license is compatible with [Apache2.0](https://github.com/ESMValGroup/ESMValTool/blob/master/LICENSE)

New recipe/diagnostic

-   [x] Add documentation for the recipe to the `doc/sphinx/source/recipes` folder and add a new entry to index.rst
-   [x] Add provenance information

Modified recipe/diagnostic

-   [ ] Update documentation for the recipe to the `doc/sphinx/source/recipes` folder
-   [ ] Update provenance information if needed
-   [ ] Assign the author(s) of the affected recipe(s) as reviewer(s)

New data reformatting script

-   [x] Test the CMORized data using recipes/example/recipe_check_obs.yml, to make sure the CMOR checks pass without errors
-   [x] Add the new dataset to the table in the [documentation](https://docs.esmvaltool.org/en/latest/input.html#supported-datasets)
-   [ ] Tag @mattiarighi in this pull request, so that the new dataset can be added to the OBS data pool at DKRZ and synchronized with CEDA-Jasmin

Modified data reformatting script

-   [ ] Test the CMORized data using recipes/example/recipe_check_obs.yml, to make sure the CMOR checks pass without errors
-   [ ] Tag @mattiarighi in this pull request, so that the updated dataset can be added to the OBS data pool at DKRZ and synchronized with CEDA-Jasmin

If you need help with any of the tasks above, please do not hesitate to ask by commenting in the issue or pull request.

* * *

Closes #issue_number

valeriupredoi commented:
excellent stuff @morobking - I assigned myself as reviewer for now and then when things are ready to get merged I'll ask @mattiarighi to have a look and induce the newly cmorized data :beer: Let's not worry about ticking the boxes just yet, functionality and code review comes first, I'll do that tomorrow :+1:

valeriupredoi commented:
@morobking two things: no need to add the config-user file you used and could you pls give me read rights on `/home/users/robking/RAWOBS` so I can run the cmorizer myself? :beer: :+1:

morobking commented:
Hi

It looks like read permission is there for all folders and files.
Let me know if there is a particular CHMOD number that would be better.

Cheers

Rob

From: Valeriu Predoi <notifications@github.com>
Sent: 04 November 2020 12:35
To: ESMValGroup/ESMValTool <ESMValTool@noreply.github.com>
Cc: King, Rob <rob.king@metoffice.gov.uk>; Mention <mention@noreply.github.com>
Subject: Re: [ESMValGroup/ESMValTool] ESA CCI LST (#1897)


This email was received from an external source.   Always check sender details, links & attachments.

@morobking<https://github.com/morobking> two things: no need to add the config-user file you used and could you pls give me read rights on /home/users/robking/RAWOBS so I can run the cmorizer myself? 🍺 👍

—
You are receiving this because you were mentioned.
Reply to this email directly, view it on GitHub<https://github.com/ESMValGroup/ESMValTool/pull/1897#issuecomment-721706316>, or unsubscribe<https://github.com/notifications/unsubscribe-auth/AQK6MLDGR6CODFSLS37KH2DSOFDAPANCNFSM4TI74AOA>.

valeriupredoi commented:
Hi Rob, cheers - I don't have `r+x` rights on that dir currently, if you make your `home` `r+x` (but not `w`) I should be able to read/copy files from you :+1: :beer:

morobking commented:
That should have done it for at least <my home>/RAWOBS

Rob.

From: Valeriu Predoi <notifications@github.com>
Sent: 04 November 2020 16:49
To: ESMValGroup/ESMValTool <ESMValTool@noreply.github.com>
Cc: King, Rob <rob.king@metoffice.gov.uk>; Mention <mention@noreply.github.com>
Subject: Re: [ESMValGroup/ESMValTool] ESA CCI LST (#1897)

NOTIFICATIONS@GITHUB.COM appears similar to someone who previously sent you email, but may not be that person. Learn why this could be a risk<http://aka.ms/LearnAboutSenderIdentification>
Feedback<http://aka.ms/SafetyTipsFeedback>

This email was received from an external source.   Always check sender details, links & attachments.

Hi Rob, cheers - I don't have r+x rights on that dir currently, if you make your home r+x (but not w) I should be able to read/copy files from you 👍 🍺

—
You are receiving this because you were mentioned.
Reply to this email directly, view it on GitHub<https://github.com/ESMValGroup/ESMValTool/pull/1897#issuecomment-721846617>, or unsubscribe<https://github.com/notifications/unsubscribe-auth/AQK6MLBPVVFJT73E4AMYWU3SOGAXBANCNFSM4TI74AOA>.

valeriupredoi commented:
@morobking sorry mate, I still don't have permissions - @agstephens could you maybe have a look pls - Rob says I should be able to read `/home/users/robking/RAWOBS` but I don't have permissions (on Jasmin) - cheers guys! :beer:

valeriupredoi commented:
@morobking excellent work with the cmorizer, mate! Cheers for migrating data on `esmeval` gws, I have succesfully run the cmorizer and I also ran a cmor check test on it and it passed fine but there are a couple minor issues: if a user will run with a strict CMOR check in esmvaltool the data will not pass it, here are the issues:

```
Standard name for lst changed from None to surface_temperature
 Long name for lst changed from land surface temperature to Surface Temperature
```

the long name is changed automatically by the tool (esmvaltool) so it's not a problem perse but it would be nice if you set it in the cmorizer; the standard_name is an issue in CMOR strict mode, so that I encourage you to set it in the cmorizer. Another suggestion would be to set the version to `1.0` rather than `1.00` (I believe there are no 0.01 precision versions right?).

Another thing I noticed is that the cmorizer consumes a lot of memory while saving the file (each yearly file, something like 14-16G), I'll have a look around in the code and see if there's any visible memory drain :+1:

morobking commented:
Thank you that good news.
I’ll sort the standard name issue out.

The version number, I found I needed to put ‘1.00’ including the quotes to get it to load the data.
The file names have 1.00 within them.
When I put this as 1.0 with no quotes, it failed.

I’m not sure what could be causing such a memory issue.

Rob.

From: Valeriu Predoi <notifications@github.com>
Sent: 09 November 2020 17:11
To: ESMValGroup/ESMValTool <ESMValTool@noreply.github.com>
Cc: King, Rob <rob.king@metoffice.gov.uk>; Mention <mention@noreply.github.com>
Subject: Re: [ESMValGroup/ESMValTool] ESA CCI LST (#1897)

NOTIFICATIONS@GITHUB.COM appears similar to someone who previously sent you email, but may not be that person. Learn why this could be a risk<http://aka.ms/LearnAboutSenderIdentification>
Feedback<http://aka.ms/SafetyTipsFeedback>

This email was received from an external source.   Always check sender details, links & attachments.

@morobking<https://github.com/morobking> excellent work with the cmorizer, mate! Cheers for migrating data on esmeval gws, I have succesfully run the cmorizer and I also ran a cmor check test on it and it passed fine but there are a couple minor issues: if a user will run with a strict CMOR check in esmvaltool the data will not pass it, here are the issues:

Standard name for lst changed from None to surface_temperature

 Long name for lst changed from land surface temperature to Surface Temperature

the long name is changed automatically by the tool (esmvaltool) so it's not a problem perse but it would be nice if you set it in the cmorizer; the standard_name is an issue in CMOR strict mode, so that I encourage you to set it in the cmorizer. Another suggestion would be to set the version to 1.0 rather than 1.00 (I believe there are no 0.01 precision versions right?).

Another thing I noticed is that the cmorizer consumes a lot of memory while saving the file (each yearly file, something like 14-16G), I'll have a look around in the code and see if there's any visible memory drain 👍

—
You are receiving this because you were mentioned.
Reply to this email directly, view it on GitHub<https://github.com/ESMValGroup/ESMValTool/pull/1897#issuecomment-724148874>, or unsubscribe<https://github.com/notifications/unsubscribe-auth/AQK6MLB2DNSQU74SAEDALXLSPAPC7ANCNFSM4TI74AOA>.

morobking commented:
Standard name is now set in the python code

From: Valeriu Predoi <notifications@github.com>
Sent: 09 November 2020 17:11
To: ESMValGroup/ESMValTool <ESMValTool@noreply.github.com>
Cc: King, Rob <rob.king@metoffice.gov.uk>; Mention <mention@noreply.github.com>
Subject: Re: [ESMValGroup/ESMValTool] ESA CCI LST (#1897)

NOTIFICATIONS@GITHUB.COM appears similar to someone who previously sent you email, but may not be that person. Learn why this could be a risk<http://aka.ms/LearnAboutSenderIdentification>
Feedback<http://aka.ms/SafetyTipsFeedback>

This email was received from an external source.   Always check sender details, links & attachments.

@morobking<https://github.com/morobking> excellent work with the cmorizer, mate! Cheers for migrating data on esmeval gws, I have succesfully run the cmorizer and I also ran a cmor check test on it and it passed fine but there are a couple minor issues: if a user will run with a strict CMOR check in esmvaltool the data will not pass it, here are the issues:

Standard name for lst changed from None to surface_temperature

 Long name for lst changed from land surface temperature to Surface Temperature

the long name is changed automatically by the tool (esmvaltool) so it's not a problem perse but it would be nice if you set it in the cmorizer; the standard_name is an issue in CMOR strict mode, so that I encourage you to set it in the cmorizer. Another suggestion would be to set the version to 1.0 rather than 1.00 (I believe there are no 0.01 precision versions right?).

Another thing I noticed is that the cmorizer consumes a lot of memory while saving the file (each yearly file, something like 14-16G), I'll have a look around in the code and see if there's any visible memory drain 👍

—
You are receiving this because you were mentioned.
Reply to this email directly, view it on GitHub<https://github.com/ESMValGroup/ESMValTool/pull/1897#issuecomment-724148874>, or unsubscribe<https://github.com/notifications/unsubscribe-auth/AQK6MLB2DNSQU74SAEDALXLSPAPC7ANCNFSM4TI74AOA>.

valeriupredoi commented:
good stuff @morobking :beer: I'll have a look at the memory issue now :+1: Could you please make sure that, when you reply to this thread but do it over email, you remove the previous messages' body (trim to your reply only, I remvoed the overheads manually) - it makes for an easier read-through :+1:

valeriupredoi commented:
@morobking cheers for making the changes - the cmorized data adheres to cmor rules 100% now :beer: As for the memory issue, I have opened a new issue for discussion with the other people and iris devvs, nothing you can do at this stage since you are using the stock functionality of the cmorizer, so don't worry about it :+1: https://github.com/ESMValGroup/ESMValCore/issues/861

morobking commented:
Thank you for that.
I’ll be placing a further pull request soon with the recipe and diagnostic scripts.

morobking commented:
@axel-lauer @valeriupredoi 
This pull request now has all the files ready for review for the CMUG ESA CCI LST diagnostic.

valeriupredoi commented:
great work @morobking - I will have a look through the code from a technical perspective tomorrow; another thing that'd be good to do is to produce the LST data with the new cmorizer and when this PR is merged to have the data migrated in `esmeval/obsdata-v2` on Jasmin and added to the list of available OBS datasets; after that we'd ask @mattiarighi to grab it on DKRZ too; @axel-lauer do you know who is @bjoernbroetz 's replacement in matters of data transfers etc - he'd normally rsync the data to and from Jasmin :beer:

valeriupredoi commented:
@morobking a few points:

- ran `yapf` on your lst.py diagnostic and automatically formatted a lot of the code to adhere to pep8 standards; there are still quite a few things to be done manually to improve the style, it would be great if you could do that please
- I commented out the `references` section in your recipe - it was empty - I put a placeholder for you there with a link on how to add references, it would be good if you created a bibtex entry as the link instructs and added that
- I would do a bit more but Jasmin has suddenly died, so that cut my work short

valeriupredoi commented:
OK Jasmin is back - a first issue I noticed with the `lst.py` script is that it was set as executable - we don't make the diag scripts executable unless they **really** need to be (almost never) - that causes runtime issues, so I unexecutabled it :grin:

morobking commented:
@valeriupredoi lst.py I've change variable names, modified strings and added docstrings to get no warnings from running prospector on the code. There is one warning remaining:
  Line: None
    pylint: failure / Tool pylint failed to run (exception was raised)

I'm not sure what this means.

I'll look at the other issues you have raised on Monday

axel-lauer commented:
> @axel-lauer do you know who is @bjoernbroetz 's replacement in matters of data transfers etc - he'd normally rsync the data to and from Jasmin 🍺

The idea is that @remi-kazeroni will take over such tasks ...

valeriupredoi commented:
cheers @axel-lauer - @remi-kazeroni I replied to your morning email, you should hopefully be in the know how to get to Jasmin :+1: 
@morobking 

> pylint: failure / Tool pylint failed to run (exception was raised)

don't worry about it, it's the pylint plugin that doesn't get run, it's fine

valeriupredoi commented:
@morobking note that your commit #950848636e4f3ca1a92b675ce2542f7eae6e4d30 has changed the file types of **all** the files in the repository, now they are all executable, which is very bad - what was your intention with that - I will have to revert that commit right now, your next commit will have to be reinstated after

valeriupredoi commented:
@morobking I don't understand why are all the files committed by you being committed as executables? Do you have some default bash setting that converts all your files on the partition you work on to executables? This is something you should be very careful not to have - it can upset a lot of core functionality down to file system level. Anyway, even the `recipe_esacci-lst.rst` was executable, I converted it to regular file, fixed the little bug in the syntax and added it to the `index` toctree - I didn't find any dedicated OBS validation section so I added it to Others - @axel-lauer maybe we should create a dedicated section for OBS recipes? :beer:

morobking commented:
I think the executable thing has happened because I was trying to make the data files in my Jasmin home space available to you before we realised that couldn’t happen.
If I run chmod 744 * will that solve this issue for next commit?

valeriupredoi commented:
@morobking try `chmod -R 744 ESMValTool` or whatever you called the ESMValTool directory and see if those files that are not supposed to be executable like the lst.py diagnostic or the recipe are still executable after - worst case just nuke the ESMValTool dir and git clone and checkout branch after :+1:

morobking commented:
@valeriupredoi Made a clean clone of the branch so shouldn’t have the executable issue now.
I’m just waiting for a reply form the LST CCI team about what is the best reference for their work; once I have that I’ll add them to the references and the recipe.
Let me know if there is anything else to look at.

valeriupredoi commented:
awesome! the only thing left to do in my books is to produce the full dataset (cmorize all the available lst data) and put it in `/group_workspaces/jasmin4/esmeval/obsdata-v2/Tier2/ESACCI-LST` please - once we've merged this PR then we can add that dataset as officially available and add it into the documentation https://docs.esmvaltool.org/en/latest/input.html#observations and ask @remi-kazeroni to rsync the data to DKRZ as well, when he'll have had set up his Jasmin account and rsync ties with DKRZ. Maybe @hb326 and/or @mattiarighi can give a once-over to this PR from the OBS nomenclature/bookkeeping point? :beer:

morobking commented:
I’ve added a reference to the ESA CCI data.
They don’t have an official reference or paper at the moment.
So this can be updated when it becomes available at a later date.

valeriupredoi commented:
good stuff, cheers @morobking - a few more things to be done before merge:
- in `lst.py`: could you either make the utility functions private (by adding `_` at the start of the function name eg `_get_input_cubes`) or add Numpy-style documentation to them, see the example I made for `diagnostic()`
- could you pls add an entry to the OBS checker [recipe](https://github.com/ESMValGroup/ESMValTool/blob/master/esmvaltool/recipes/examples/recipe_check_obs.yml) for LST - see the other entries for an example, eg SST
- would you be ok to generate the cmorized data and put it somewhere on Jasmin so I can migrate it in esmeval pls?

@axel-lauer @hb326 yet another cmorizer for your attention as scientific reviewers, guys, cheers :beer:

axel-lauer commented:
I was just about to test this new cmorizer but I am missing the downloading instructions. I think giving a workspace on Jasmin to get the input files is not a good option as not everyone has access and because files might be removed from there at any time.

valeriupredoi commented:
and a very good point @axel-lauer - cheers for looking into this! Yes, this PR needs a bit more work to be ready for a merge; but it's almost there - @morobking could you please address Axel's comment while you're working on the tweaks I suggested in my [comment](https://github.com/ESMValGroup/ESMValTool/pull/1897#issuecomment-743159514) above? Don't have to be this year, in fact, I will be officially on Xmas break from mid-day today myself, happy holidays guys! :santa: :christmas_tree:

morobking commented:
I’ll take a look for where the original files are, I’ll need to confirm them with the CCI team.
Where would be a good place to put output from the CMORizer so you can access it? Is there a good place on Jasmin?

I am having a few technical issues running at the moment, so hopefully I resolve these soon.

One question now, to add “add an entry to the OBS checker” do I only need to edit this file? ESMValTool<https://github.com/ESMValGroup/ESMValTool>/esmvaltool<https://github.com/ESMValGroup/ESMValTool/tree/master/esmvaltool>/recipes<https://github.com/ESMValGroup/ESMValTool/tree/master/esmvaltool/recipes>/examples<https://github.com/ESMValGroup/ESMValTool/tree/master/esmvaltool/recipes/examples>/recipe_check_obs.yml

The workflow and code for this is forming the base of my CMUG WP5.4 work, issue #1945
So hopefully as long as we’re happy with it scientifically, then the pull request process should be smoother.

axel-lauer commented:
Yes, in order to “add an entry to the OBS checker” it is only needed to add the new dataset to recipe_check_obs.yml.

morobking commented:
@axel-lauer 
The data is currently only available on Jasmin here:
/group_workspaces/jasmin2/esacci_lst/public/AQUA_MODIS_L3C/1.00/YYYY/MM/00
where YYYY is the 4 digit year 2002-2018, and MM is the month number with leading zero.
Each folder has two files
ESACCI-LST-L3C-LST-MODISA-0.05deg_1MONTHLY_DAY-YYYYMM00000000-fv1.00.nc
ESACCI-LST-L3C-LST-MODISA-0.05deg_1MONTHLY_NIGHT-YYYYMM00000000-fv1.00.nc
with YYYY and MM as above.

valeriupredoi commented:
@axel-lauer cheers for the ongoing review and @morobking cheers for the ongoing work - it would be quite nice if this made through to v2.2 release - you guys reckon all will be settled by the code freeze deadline of Feb 15?

valeriupredoi commented:
> @axel-lauer cheers for the ongoing review and @morobking cheers for the ongoing work - it would be quite nice if this made through to v2.2 release - you guys reckon all will be settled by the code freeze deadline of Feb 15?

friendly ping @morobking @axel-lauer - we're still accepting PR's into for the release, guys, last orders soon :beer:

morobking commented:
Sure, just let me know what else needs doing form me.

valeriupredoi commented:
@morobking I'll have one more look tomorrow but from a technical stand point it should be fine, a few more points that @axel-lauer was raising in this [comment](https://github.com/ESMValGroup/ESMValTool/pull/1897#issuecomment-754057216) - oh and remember that esmeval gws changed location on JASMIN at `/gws/nopw/j04/esmeval/obsdata-v2`

valeriupredoi reviewed (approved):
good job @morobking - nice additions to the cmorizing collection, cheers :beer: I am approving this PR from a technical point **BUT** please do this before committing any more changes: since your source code dir seems to still be fully chmodded to executable (which is very dangerous if committed to remote source), whenever you commit a change to, say, file FILE, run:
```
chmod -x FILE
git commit FILE -m "your commit message"
```
This is very important since introducing different file permissions on files that dont need that will lead to unexpected runtime behaviour.

Another point is the one raised by @axel-lauer - currently the data is only available from a private group workspace on JASMIN that I can't access myself, so please make the raw data available from a public source and update the source in the cmorizer script :beer:

valeriupredoi commented:
@axel-lauer @remi-kazeroni when you guys are happy with the data sourcing and running the cmorizer, please don't merge, just ping me and I will do a last check on the files see if they are properly cmodded to `-x` and then I will merge :beer:

morobking commented:
I need to check with the LST CCI team about when the data will be available more widely.

morobking commented:
I have this reply from the CCI team:
I don’t expect any further major updates now to what is available on Jasmin until the Autumn, when the data are at the stage of the first full release and should also become available via the CCI data portal.  The products on the Jasmin open data portal are Beta versions, but it is possible these may be updated, at least in part, with more up-to-date versions of the data as they are processed ready for the full release.

valeriupredoi commented:
cheers for inquiring and pushing on with this @morobking - I am a bit uneasy to let a cmorizer get to become public if the actual data is not public (to a certain extent) but given the data is actually available on JASMIN, under some restrictions, I might say this could go into master, but am on the fence - what do you think @axel-lauer

axel-lauer commented:
@morobking thanks for checking the data availability again. I agree with @valeriupredoi that it might make sense to wait with the merging until the ESA dataset is publicily available if we keep everything as is.

Alternatively, we could declare the current dataset as "tier3" data (restricted data sets). The data would then be only available to the ESMValTool development team on Jasmin and Mistral (DKRZ) but not to external people. Once the ESA data have been officially released, we can update the CMORizer and move the data back to "tier 2" (other freely available data sets). Would that be an option?

morobking commented:
I’ll send that message to the CCI team, but I think that will be fine.
For me, whatever is best for ESMValTool at the moment is fine.

To note, we will have this same issue for the diagnostic I’m working on for CMUG WP5.4 as it uses the same CCI data files.

axel-lauer commented:
I tested the ESACCI-LST CMORizer and the new LST diagnostic. The output from both, the CMORizer and the diagnostic, look scientifically fine but I noticed a few things that I think would be good to address before merging (see below).

## CMORIzer (cmorize_obs_esacci_lst.py)
* produces files hat are named "..._Amon_ts_..." but which actually contain a variable called "lst" instead of "ts".
* start_year and end_year in the config file (cmor_config/ESACCI-LST.yml) are ignored
* information on mip and CMOR tables seem not to be used by the CMORizer

@valeriupredoi Could you maybe check? Adding the following code in line 86 of the CMORizer script creates output with variable name "ts", but CMOR tables are still ignored I guess. Not sure that's a problem or not.
```
            this_years_cubes.var_name = 'ts'
```

## Diagnostic (lst/lst.py)
* beause of https://github.com/ESMValGroup/ESMValCore/issues/1096 I had to add the following lines to the preprocessor function "mask_landsea" in "recipe_eascci_lst.yml" (otherwise recipe would crash):
```
      fx_variables:
        sftlf:
```
* output filetype 'png' is hardwired (line 113), i.e. output_file_type defined in user-config.yml is ignored
* ESACCI-LST seems to be hardwired (e.g. line 189), maybe use attribute "reference_dataset" from recipe instead?
* runnnig the recipe gives the following warnings because of undefined references:
```
WARNING [37607] The reference file /mnt/lustre02/work/bd0854/b380103/ESMValTool/esmvaltool/references/lst.bibtex does not exist, citation information incomplete.
WARNING [37607] The reference file /mnt/lustre02/work/bd0854/b380103/ESMValTool/esmvaltool/references/esacci.bibtex does not exist, citation information incomplete.
```

valeriupredoi commented:
cheers @axel-lauer - having a look now, good catches :+1:

valeriupredoi commented on esmvaltool/cmorizers/obs/cmorize_obs_esacci_lst.py:
```diff
@@ -0,0 +1,154 @@
+
+Source
+   On CEDA-JASMIN
+   /group_workspaces/jasmin2/esacci_lst/public
```
```suggestion
   /gws/nopw/j04/esacci_lst/public
```

valeriupredoi commented on esmvaltool/cmorizers/obs/cmorize_obs_esacci_lst.py:
```diff
@@ -0,0 +1,154 @@
+   On CEDA-JASMIN
+   /group_workspaces/jasmin2/esacci_lst/public
+
+Download and processing instructions
```
also it'd be nice to say that one needs to register for the JASMIN group workspace, otherwise the data is unavailable, registration can happen at https://accounts.jasmin.ac.uk/services/group_workspaces/esacci_lst/

axel-lauer replied:
Added some info on the registration! Thank for catching this.

valeriupredoi commented:
unfortunately I can't test with real data (maybe @remi-kazeroni has access to the data and he can test?) but it seems you are absolutely correct @axel-lauer - output cube's `short_name, long_name` and `standard_name` are not set in the CMORizer (I don't know how I did the previous review, I assumed they don't change, or I glossed over it too quickly, my bad). @morobking have a look at the e.g. [GRACE](https://github.com/ESMValGroup/ESMValTool/blob/39764f4abf495a1e150dc9dd3851752c38e18f79/esmvaltool/cmorizers/obs/cmorize_obs_grace.py#L118) CMORizer how it's done , including the use of the CMOR table to extract the relevant info for correct variable setting :+1:

valeriupredoi commented:
Axel's points:

> - output filetype 'png' is hardwired (line 113), i.e. output_file_type defined in user-config.yml is ignored

yup, you are correct, this should be changed to a more flexible config-defined extension but it's no biggie for now

> - ESACCI-LST seems to be hardwired (e.g. line 189), maybe use attribute "reference_dataset" from recipe instead?

yeah, but this is a very specific diagnostic anyway, I'd argue it kind of needs a lot more flexibility to start with, this change would be a good start to make it more flexible :+1: 
> - runnnig the recipe gives the following warnings because of undefined references

yeah I can't see the references either, Rob could you please add them? Here's a guide how to [add references](https://docs.esmvaltool.org/en/latest/community/diagnostic.html#adding-references)

axel-lauer commented:
@morobking @valeriupredoi I am happy to run the CMORizer on Mistral and take a look at the output once the proposed changes changes have been applied.

valeriupredoi commented:
@morobking could you please have a look at this and what's needed before a final test from @axel-lauer and/or @remi-kazeroni when you have time, mate :beer:

morobking commented:
@axel-lauer is this what you emailed me about Axel?
Two thoughts:
Its not clear to me what need to be done from looking back at this conversation here, can you point it out please.

Also, when I have tried to switch between the CMUG WP5.3 and WP5.4 before, it lead to problems about not getting the right version of ESMValCore (the WP5.4 work modifies Core as well as Tool). Is this indicative of a wider problem, or likely to be something I'm doing wrong?
Cheers.

axel-lauer commented:
> @axel-lauer is this what you emailed me about Axel?

I think getting this PR ready for merging would be what we need for our CMUG deliverable.

> Two thoughts: Its not clear to me what need to be done from looking back at this conversation here, can you point it out please.

Maybe you could revisit the comments on the cmorizer and the diagnostic summarized here
https://github.com/ESMValGroup/ESMValTool/pull/1897#issuecomment-845847164
and check whether these issues are resolved or not relevant (any more). If so, then please let me know and I will run some tests before merging. If not, please try to answer the points raised so we know where we are. 

> Also, when I have tried to switch between the CMUG WP5.3 and WP5.4 before, it lead to problems about not getting the right version of ESMValCore (the WP5.4 work modifies Core as well as Tool). Is this indicative of a wider problem, or likely to be something I'm doing wrong?

Not sure what could be going wrong. I think I would recommend to start with a fresh installation using Mamba explained here:
https://docs.esmvaltool.org/en/latest/quickstart/installation.html#install-with-mamba
and then try again.

morobking commented:
@axel-lauer @valeriupredoi  
Thanks Axel for highlighting this thread back to me.
If I understand Axel's comments from above, this is the list of things to do:

- sort an issue with variable name ts vs lst. @valeriupredoi can you please let me know what to do following Axel's comment re CMOR tables being ignored
-    Issues with preprocessors that use fx variables ESMValCore#1096 I had to add the following lines to the preprocessor function "mask_landsea" in "recipe_eascci_lst.yml" (otherwise recipe would crash):
      fx_variables:
        sftlf:
 Is this as simple as adding those lines to the yml file? I dont understand what this is doing.
- output file type, I couldnt get the built in plotting commands to work, hence whyin the .py file I used matplotlib the way I have. Isnt plot type given inthe user config file? But in the other diagnostic I'm working on I've just had to remoe this option from the config file to allow the lastest version of esmvaltool to work, and cannot see where to place. I'm not 100% sure how to implement that, unless there is a way of getting that option into my .py file and I can make the plt.savefig command have it as an option inthe filename string.

 It doesnt seem to me that there is much to do, unless I've really misunderstood this, I just need some pointers in the right direction please.

axel-lauer commented:
@morobking Thanks for looking into this!

> sort an issue with variable name ts vs lst. @valeriupredoi can you please let me know what to do following Axel's comment re CMOR tables being ignored

@valeriupredoi would be great if you could take a brief look.

> Issues with preprocessors that use fx variables ESMValCore#1096 I had to add the following lines to the preprocessor function "mask_landsea" in "recipe_eascci_lst.yml" (otherwise recipe would crash):
>       fx_variables:
>       sftlf:
>       Is this as simple as adding those lines to the yml file? I dont understand what this is doing.

I think this is resolved by now, so nothing to do from your side.

> output file type, I couldnt get the built in plotting commands to work, hence whyin the .py file I used matplotlib the way I have. Isnt plot type given inthe user config file? But in the other diagnostic I'm working on I've just had to remoe this option from the config file to allow the lastest version of esmvaltool to work, and cannot see where to place. I'm not 100% sure how to implement that, unless there is a way of getting that option into my .py file and I can make the plt.savefig command have it as an option inthe filename string.

I think you would need to use the extension specified in config-user.yml via "output_file_type" when calling 'savefig' in lst.py. At the moment, '.png' is hard coded as extension, i.e. the function will always create a .png.

> It doesnt seem to me that there is much to do, unless I've really misunderstood this, I just need some pointers in the right direction please.

Yes, I think we are getting very close to be able to merge this!

Here is some little things I changed (hope that was fine):
- added a title to the recipe
- changed reference in recipe from "esacci-lst" to "esacci_lst" (replaced hyphen by underscore) so the refence name is interpreted correctly by yaml
- added info on registration for access to JASMIN workspace to CMORizer
- updated path to JASMIN workspace as suggested by @valeriupredoi 
- changed preprocessor multi_model_statistics from "std" to "std_dev" (same in diagnostic)

axel-lauer reviewed (approved):
Would be nice if you could replace the hard coded ".png" when creating the plot in the diagnostic with the setting specified in config-user.yml (output_file_type). Other than that, I think this is ready to be merged.
